csharp/ql/src/semmle/code/csharp/dataflow/internal/DelegateDataFlow.qll

```diff
@@ -85,33 +106,49 @@ abstract private class DelegateFlowSink extends DataFlow::Node {
    */
   cached
   Callable getARuntimeTarget(CallContext context) {
-    exists(DelegateFlowSource dfs |
+    exists(DelegateLikeFlowSource dfs |
       flowsFrom(this, dfs, _, context) and
       result = dfs.getCallable()
     )
   }
 }
 
+/** A delegate or function pointer call expression. */
+class DelegateLikeCallExpr extends DelegateLikeFlowSink, DataFlow::ExprNode {
```

Comment:
I think you should be able to define this directly via `DelegateLikeCall`, so no need for the sub classes `DelegateCallExpr` and `FunctionPointerCallExpr`.